common/datatypes.py

```diff
@@ -2794,14 +2794,14 @@ def set_menu_data(self, class_menu_data):
 
         return self
 
-    def set_player(self, unique_id, name, tag, loadouts):
+    def set_player(self, unique_id, name, tag, loadouts, game_setting_mode: str):
         self.findbytype(m0348).set(unique_id)
         self.findbytype(m034a).set(name)
         self.findbytype(m06de).set(tag)
 
         loadout_arrays = []
         loadout_overall_idx = 0
-        for class_id, class_loadout in loadouts.loadout_dict.items():
+        for class_id, class_loadout in loadouts.get_loadouts(game_setting_mode).items():
```

Comment:
Have added methods to the player to hide this. The method on the player still calls the method on the loadouts with the parameter, since I don't want to duplicate the state in both the player and loadouts object.

But external calls (like this) no longer need to provide it, it will get the loadouts for the correct mode.

Reply:
Exactly, that's what I meant.